src/coreclr/tools/Common/TypeSystem/Interop/IL/Marshaller.Aot.cs

```diff
@@ -913,17 +913,29 @@ protected override void AllocAndTransformNativeToManaged(ILCodeStream codeStream
         {
             ILEmitter emitter = _ilCodeStreams.Emitter;
 
-            MethodDesc helper = Context.GetHelperEntryPoint("InteropHelpers", "ConvertNativeComInterfaceToManaged");
             LoadNativeValue(codeStream);
 
+            MethodDesc helper = Context.GetHelperEntryPoint("InteropHelpers", "ConvertNativeComInterfaceToManaged");
             codeStream.Emit(ILOpcode.call, emitter.NewToken(helper));
 
             StoreManagedValue(codeStream);
         }
 
         protected override void TransformNativeToManaged(ILCodeStream codeStream)
         {
-            throw new NotSupportedException();
+            ILEmitter emitter = _ilCodeStreams.Emitter;
+            ILCodeLabel lNull = emitter.NewCodeLabel();
+
+            LoadNativeValue(codeStream);
+            codeStream.Emit(ILOpcode.brfalse, lNull);
+
+            LoadNativeValue(codeStream);
+            MethodDesc helper = Context.GetHelperEntryPoint("InteropHelpers", "ConvertNativeComInterfaceToManaged");
+            codeStream.Emit(ILOpcode.call, emitter.NewToken(helper));
+
+            StoreManagedValue(codeStream);
+
+            codeStream.EmitLabel(lNull);
```

Comment:
Is it correct to skip storing the managed value for null?

Reply:
I think yes. At least this pattern is everywhere.

Reply:
Managed value initialized as `null` by runtime guarantee, so load/store usually going in pairs. Then somebody do actual return. At least this is how I understand this.

Reply:
> Managed value initialized as null by runtime guarantee,

Where is it guaranteed?

I went through all TransformNativeToManaged methods and all of them end with Store. I have not found one that skips the store.

Reply:
LayoutClassPtrMarshaller, AsAnyMarshaller for example

Reply:
The managed value for these marshallers is a pointer to the structure that they are marshalling. It is not the case here.

Reply:
What I was trying to say that
```
.method public hidebysig specialname 
        instance int32  get_Set() cil managed
{
  .maxstack  1
  .locals init (int32 V_0)
  IL_0009:  ldloc.0
  IL_000a:  ret
} // end of method SetProperty::get_Set

```

is equivalent for 
```
int Set => 0;
```

Reply:
I miss your previous message. Will have to check what kind of IL actually generated to validate that what I'm saying is correct.

Reply:
You may be right, but I would still simplify this to a simple ConvertNativeComInterfaceToManaged. If the store is redudandant, the JIT will eliminate it after inlining ConvertNativeComInterfaceToManaged.